torch/optim/rmsprop.py

```diff
@@ -369,6 +421,14 @@ def _multi_tensor_rmsprop(
         if momentum > 0:
             torch._foreach_mul_(grouped_momentum_buffer_list, momentum)
             torch._foreach_addcdiv_(grouped_momentum_buffer_list, grouped_grads, avg)
-            torch._foreach_add_(grouped_params, grouped_momentum_buffer_list, alpha=-lr)
+            if capturable:
+                momentum_lr = torch._foreach_mul(grouped_momentum_buffer_list, -lr)
+                torch._foreach_add_(grouped_params, momentum_lr)
+            else:
+                torch._foreach_add_(grouped_params, grouped_momentum_buffer_list, alpha=-lr)
         else:
-            torch._foreach_addcdiv_(grouped_params, grouped_grads, avg, value=-lr)
+            if capturable:
+                torch._foreach_div_(avg, -lr)
```

Comment:
```suggestion
                torch._foreach_mul_(grouped_grads, -lr)
```
why not mul? i think mul should be faster than div

Reply:
We actually had larger error when I did that - I don't think the grads are properly copied in some configurations. This shouldn't affect performance anyway as these are memory-bound kernels.

Reply:
What do you mean the grads are not properly copied?

Reply:
Basically I think there are cases above where we are modifying the actual grads (if we aren't maximize, no weight decay), but regardless, the error is higher, and there is no performance degradation from this, so I think the div is fine.

Reply:
this is concerning cuz the grads should not get modified in the update. This does not have to do with your code so no action needed here but if you’re noticing bad stuff elsewhere pls flag